### PR TITLE
[KernelGen][Nvidia] Move hardtanh operator to ops

### PR DESCRIPTION
**Pull request by YufanPeter**
### Summary
  Move `hardtanh` operator (`hardtanh` and `hardtanh.out`) from `experimental_ops` to
  `ops`.

  ### Testing
  - Parametrized correctness tests for `hardtanh` and `hardtanh.out` across multiple shapes and all
  supported float dtypes, including explicit `(min_val, max_val)` ranges
  - Covered special values (±inf / ±0 / NaN), verifying NaN is propagated to match PyTorch semantics
  - Validated against the PyTorch reference with `--ref cpu`
  - Tested on: NVIDIA H20

  ### Performance
  Tested on NVIDIA H20 using:
  ```bash
  pytest -s benchmark/test_hardtanh.py --level core
  ```

  #### hardtanh

  **float16**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 1.435 | 1.602 | 0.895x |
  | [64, 64] | 0.006 | 0.005 | 1.012x |
  | [4096, 4096] | 0.028 | 0.024 | 1.170x |
  | [64, 512, 512] | 0.028 | 0.024 | 1.168x |
  | [1024, 1024, 1024] | 1.475 | 1.259 | 1.172x |

  **float32**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 2.053 | 2.027 | 1.013x |
  | [64, 64] | 0.005 | 0.006 | 0.983x |
  | [4096, 4096] | 0.037 | 0.038 | 0.989x |
  | [64, 512, 512] | 0.038 | 0.038 | 0.996x |
  | [1024, 1024, 1024] | 2.077 | 2.037 | 1.020x |

  **bfloat16**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 1.404 | 1.190 | 1.179x |
  | [64, 64] | 0.005 | 0.006 | 0.983x |
  | [4096, 4096] | 0.026 | 0.024 | 1.093x |
  | [64, 512, 512] | 0.026 | 0.024 | 1.078x |
  | [1024, 1024, 1024] | 1.416 | 1.223 | 1.158x |

  #### hardtanh_out

  **float16**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 1.513 | 1.203 | 1.257x |
  | [64, 64] | 0.005 | 0.005 | 1.021x |
  | [4096, 4096] | 0.027 | 0.023 | 1.162x |
  | [64, 512, 512] | 0.027 | 0.024 | 1.147x |
  | [1024, 1024, 1024] | 1.503 | 1.272 | 1.182x |

  **float32**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 2.069 | 2.046 | 1.011x |
  | [64, 64] | 0.006 | 0.005 | 1.054x |
  | [4096, 4096] | 0.037 | 0.040 | 0.941x |
  | [64, 512, 512] | 0.038 | 0.039 | 0.958x |
  | [1024, 1024, 1024] | 2.062 | 2.056 | 1.003x |

  **bfloat16**
  | shape | Torch latency | Gems latency | speedup |
  |---|---|---|---|
  | [1073741824] | 1.400 | 1.196 | 1.170x |
  | [64, 64] | 0.005 | 0.005 | 1.037x |
  | [4096, 4096] | 0.026 | 0.023 | 1.108x |
  | [64, 512, 512] | 0.026 | 0.024 | 1.092x |
  | [1024, 1024, 1024] | 1.381 | 1.202 | 1.149x |

  #### Average
  Combined average speedup: **1.073x**

  ### Files changed
  - `src/flag_gems/ops/hardtanh.py`: Production Triton kernel for `hardtanh` and `hardtanh.out`
  - `tests/test_hardtanh.py`: Accuracy tests for `hardtanh` / `hardtanh.out`
  - `benchmark/test_hardtanh.py`: Performance benchmark using the standard benchmark framework
  - `src/flag_gems/ops/__init__.py`: Export `hardtanh` / `hardtanh_out` through imports and `__all__`
  - `src/flag_gems/__init__.py`: Register `hardtanh` and `hardtanh.out` in the aten patch list
  - `conf/operators.yaml`: Add `hardtanh` / `hardtanh.out` entries `(kind: NeuralNetwork, stage: alpha 5.4)`
  - `src/flag_gems/experimental_ops/hardtanh.py`: Removed after migration to `ops`
  - `experimental_tests/unit/hardtanh_test.py`: Removed after migration to `ops`
  - `experimental_tests/performance/hardtanh_test.py`: Removed after migration to `ops`

### CODE REVIEW

File: tests/test_hardtanh.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        -2.0,
+    ]
+    res_inp = torch.tensor(values, dtype=dtype, device=flag_gems.device)
+    ref_inp = utils.to_reference(res_inp, True)
+
+    ref_out = torch.ops.aten.hardtanh(ref_inp)
+    with flag_gems.use_gems():
+        res_out = torch.ops.aten.hardtanh(res_inp)
+
+    utils.gems_assert_close(res_out, ref_out, dtype, equal_nan=True)
```
Comment (bin913) [inline tests/test_hardtanh.py:116]: Please add hardtanh_out  in accuracy test

File: tests/test_hardtanh.py [hunk 2]
```diff
@@ -114,3 +114,34 @@ def test_hardtanh_special_values(dtype):
         res_out = torch.ops.aten.hardtanh(res_inp)
 
     utils.gems_assert_close(res_out, ref_out, dtype, equal_nan=True)
+
+
+@pytest.mark.hardtanh
```
Comment (bin913) [inline tests/test_hardtanh.py:119]: @pytest.mark.hardtanh_out
